src/WebJobs.Script/Workers/Rpc/SharedMemoryDataTransfer/SharedMemoryManager.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO.MemoryMappedFiles;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.MemStore.MemoryMappedFileHandler.ResponseTypes;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    internal class SharedMemoryManager
```

Comment:
New type `SharedMemoryFile` should handle methods exposed here as well.

Reply:
I want to keep this separate as this is going to act as a basic collection/manager of `SharedMemoryFile`s. It holds the references (to keep the allocated memory) until it is used (after the invocation is over) and then can free the `SharedMemoryFile`s and the allocated memory.